.github/workflows/rust-windows.yml

```diff
@@ -0,0 +1,57 @@
+name: Rust CI (Windows)
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: ["*"]
+
+jobs:
+  build-and-test:
+    runs-on: windows-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+
+      - name: Cache cargo registry
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Install MongoDB
+        run: choco install mongodb --version 4.4.6
+
+      - name: Create MongoDB data and log directories
+        run: |
+          New-Item -ItemType Directory -Path C:\data\db -Force
+          New-Item -ItemType Directory -Path C:\data\log -Force
+
+      - name: Start MongoDB Service
+        run: |
+          Set-Service -Name mongodb -StartupType Automatic
+          Start-Service -Name mongodb
```

Comment:
This should not be necessary. It is the responsibility of `temp-mongo` to configure, initialize and start `mongod`.